core/common/src/main/java/alluxio/heartbeat/CronExpressionIntervalSupplier.java

```diff
@@ -56,4 +57,22 @@ public long getRunLimit(long mPreviousTickedMs) {
     return Duration.between(now.toInstant(),
         mCron.getNextInvalidTimeAfter(now).toInstant()).toMillis();
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    CronExpressionIntervalSupplier that = (CronExpressionIntervalSupplier) o;
+    return mInterval == that.mInterval
+        && Objects.equals(mCron.getCronExpression(), that.mCron.getCronExpression());
```

Comment:
need a null check on `that.mCron`. I think you can add a `Preconditions.notNull(cronExpression, "CronExpression is null")` in the constructor

Reply:
fix.